plone/formwidget/namedfile/datamanager.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+from plone.formwidget.namedfile.interfaces import IScaleGenerateOnSave
+from plone.namedfile.field import INamedImageField
+from z3c.form.datamanager import AttributeField
+from ZODB.POSException import ConflictError
+from zope.annotation.interfaces import IAnnotations
+from zope.component import adapter
+from zope.component import getMultiAdapter
+from zope.component import getUtility
+from zope.globalrequest import getRequest
+from zope.interface import Interface
+from zope.interface import alsoProvides
+
+import logging
+import os
+import transaction
+
+try:
+    from Products.CMFPlone.factory import _IMREALLYPLONE5  # noqa
+except ImportError:
+    PLONE_5 = False  # pragma: no cover
+else:
+    PLONE_5 = True  # pragma: no cover
+
+
+ANNOTATION_KEY = "plone.formwidget.namedfile.scale"
+ENVIRONMENT_KEY = "PLONE_SCALE_GENERATE_ON_SAVE"
+
+logger = logging.getLogger(__name__)
+
+
+@adapter(Interface, INamedImageField)
+class NamedImageAttributeField(AttributeField):
+
+    def __init__(self, *args, **kwargs):
+        super(NamedImageAttributeField, self).__init__(*args, **kwargs)
+        flag = (os.environ.get(ENVIRONMENT_KEY) or "").lower()
+        self.scale_generate_on_save = flag in ["1", "true", "yes", "on"]
+
+    def set(self, value):
+        """See z3c.form.interfaces.IDataManager"""
+        super(NamedImageAttributeField, self).set(value)
+        if self.scale_generate_on_save:
+            schedule_plone_scale_generate_on_save(
+                self.context, getRequest(), self.field.__name__)
+
+
+def schedule_plone_scale_generate_on_save(context, request, fieldname):
```

Comment:
I'm thinking if both `schedule_plone_scale_generate_on_save` and  `plone_scale_generate_on_save` should go to utils instead.

Reply:
As you wish!
My comment was just a suggestion not a requirement!
Every time I see some function using a context and a request it seems to me it looks like a view method, but if the circumstances require it to be an helper utility no problem :)

Thanks for your time!

Reply:
`schedule_plone_scale_generate_on_save` really looks like it could be adapter by its own, but maybe do that only when there's need for a such flexibility.

I'll move those into utilities. No more too many functions in datamanager.py :)

Reply:
And, at the end, I decided to leave it as it was. I feel that right now datamanager.py tells in chronological order how this feature works.